staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -114,6 +114,9 @@ func NewNamedReflector(name string, lw ListerWatcher, expectedType interface{},
 		period:        time.Second,
 		resyncPeriod:  resyncPeriod,
 		clock:         &clock.RealClock{},
+		// We set lastSyncResourceVersion to "0", because it's the value which
+		// we set as ResourceVersion to the first List() request.
+		lastSyncResourceVersion: "0",
```

Comment:
there are places that implement `hasSynced` as `return reflector.LastSyncResourceVersion() != ""`

I don't think we should modify the publicly visible behavior of LastSyncResourceVersion to track the resourceVersion we should relist at

Reply:
Good catch. I've introduced `relistResourceVersion` and reverted `lastSyncResourceVersion` back to initializing to an empty string.